orca-core/src/main/groovy/com/netflix/spinnaker/orca/RetryableTask.java

```diff
@@ -1,10 +1,16 @@
 package com.netflix.spinnaker.orca;
 
+import java.time.Duration;
+
 /**
  * A retryable task defines its backoff period (the period between delays) and its timeout (the total period of the task)
  */
 public interface RetryableTask extends Task {
   long getBackoffPeriod();
 
   long getTimeout();
+
+  default long getExponentialBackoffPeriod(Duration taskDuration) {
```

Comment:
You're right (was originally thinking of doing something more generic across the board but pulled back) ... perhaps `getDynamicBackoffPeriod()` is better?